libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -341,13 +341,43 @@ void AP_Camera::send_feedback(mavlink_channel_t chan)
         0.0f, CAMERA_FEEDBACK_PHOTO, _camera_trigger_logged);
 }
 
+void AP_Camera::update_campos() {
+    if (_campos_updated_roll_angle) {
+        return;
+    }
+    if (_campos_positions < 2) {
+        return;
+    }
+
+    uint32_t tnow = AP_HAL::millis();
+    if (_last_photo_time + _min_interval > tnow) {
+        return;
+    }
+
+    if (++_campos_pose_counter == _campos_positions) {
+        _campos_pose_counter = 0;
+    }
+
+    float roll = _campos_angle_interval * _campos_pose_counter - _campos_roll;
+    _campos_updated_roll_angle = true;
+
+    AP_Mount *mount = AP_Mount::get_singleton();
+    if (!mount) {
+        return;
+    }
+    
+    mount->set_angle_targets(roll, _campos_pitch, 0);
```

Comment:
I think this is the biggest question.. we've never had the camera controlling the mount before

Reply:
I implemented it this way as I got inspiration from the CAM_TRIGG_DIST, which is the closest command that I could relate. In fact, in the original PR at mavlink/mavlink it was initially considered to use params 4~6 of CAM_TRIGG_DIST to implement this feature.
I understand your concern about having the mount controlled by the camera as it is a slightly new concept, but I thought it would be reasonable, as it is not so different from issuing mavlink commands to other components that are not implemented in AP_Camera, as it is done in a few places.
Lastly, I had the peace of mind that the sole purpose of a mount is to hold and stabilize/point the camera to the desired target, and that a mount-enabled vehicle must have a camera, in this sense, I could relate the camera/mount relationship to a master/slave system.

https://github.com/ArduPilot/ardupilot/blob/5edae526fda9ac6637c3fc9eafaea5fc967ce41a/libraries/AP_Camera/AP_Camera.cpp#L270
https://github.com/ArduPilot/ardupilot/blob/5edae526fda9ac6637c3fc9eafaea5fc967ce41a/libraries/AP_Camera/AP_Camera.cpp#L315
https://github.com/ArduPilot/ardupilot/blob/5edae526fda9ac6637c3fc9eafaea5fc967ce41a/libraries/AP_Camera/AP_Camera.cpp#L482